### PR TITLE
fix(form-field): fix underline at different zoom levels

### LINKED ISSUE #7494: (Bug) .mat-form-field-underline not rendering in Webkit browsers
First off -- thanks for the all the great code!

#### Bug, feature request, or proposal:

The line underneath the input in a form field (any type of form field) with the class `.mat-form-field-underline` is sometimes absent from Webkit browsers. It renders as expected at any zoom level and any size screen or device on Firefox, but with Chrome and Safari it's completely random. For example, I have a list in my application using the plain ol' input and datepicker elements. Some of them have underlines, some of them don't. It's different which ones are working and which aren't on any given render. 

I thought it may have something to do with the height being changed to `0.99px` in Chrome, but even setting the height to `1.1px !important` using `::ng-deep` doesn't fix it permanently. Something that may help is that when I use that `1.1px` height trick, some lines look extra thick, while others just look normal. I would think maybe those normal looking ones are the ones that would be absent without explicitly setting the height, but then again, sometimes they disappear altogether.

#### What is the expected behavior?

The underlines would show up all the time, no matter what.

#### What is the current behavior?

Unpredictable rendering for underlines.

#### What are the steps to reproduce?
https://plnkr.co/edit/kmjxWTwC6sSrkSnDVoGQ?p=preview

Check out this Plunker on Chrome and zoom in and out, resize your window, refresh the page a bunch of times, eventually the underline will disappear. For me, 80% is a pretty reliable way to make it disappear, but I'm sure it's different everywhere (different screens, devices, etc.). I wish I could provide more info, but it is truly random. 

#### What is the use-case or motivation for changing an existing behavior?

Using the input. : /

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Webkit browsers.

#### Is there anything else we should know?
Again, great library. Thanks!

### PR DESCRIPTION
**Pull request by mmalerba**
fixes #7494

### CODE REVIEW

File: src/lib/form-field/form-field.scss [hunk 1]
```diff
@@ -145,6 +145,8 @@ $mat-form-field-underline-height: 1px !default;
   position: absolute;
   height: $mat-form-field-underline-height;
   width: 100%;
+  // Prevents underline from disappearing at lower zoom levels.
+  transform: perspective(1px);
```
Comment (crisbeto) [inline src/lib/form-field/form-field.scss]: I'm not sure whether this fixes the issue. The only way I managed to break it on my machine was to zoom out to 33% and these changes don't seem to help in that case.

Comment (mmalerba) [reply src/lib/form-field/form-field.scss]: Its a little weird, they don't all seem to break, I think it has to do with their vertical position or something. I saw some of them broken at 90% on the input demo page, this fixes it for >= 50% which I think is reasonable

Comment (crisbeto) [reply src/lib/form-field/form-field.scss]: I think that it all depends on your OS/Screen size/Browser combo and whether it hits some weird value that doesn't round off correctly. We could get this in and see if people still see the issue.

Comment (mmalerba) [reply src/lib/form-field/form-field.scss]: Yeah, I suspect this will fix it, it does for me

**Comment by jecastrops** [comment]
This fix is ​​great but at 33% and 25% zoom the line disappears again

**Pull request merged**